### DISCUSSION TITLE
Documentation for mitmdump script arguments

### DISCUSSION THREAD

**Discussion by halfbusy** [mitmproxy/mitmproxy#6444] [Q&A]
The follow example from the MITM documentation site  appears to take an argument from the CLI. 

`Writer(sys.argv[1])`

https://docs.mitmproxy.org/stable/addons-examples/#io-write-flow-file

However, I cannot find appropriate documentation on how to execute this example using with mitmdump.

The best documentation I can find is from 2017, but doesn't work on the current release (errors documented further below)
`>>> mitmdump -dd -s "./arguments.py html fakehtml" -r saved -w changed`
https://github.com/mitmproxy/mitmproxy/blob/aa0e1340417385302a8c6f5034f9151aadbc25b5/docs/scripting/overview.rst#handling-arguments

Reproduction below:

I inserted a logger to check the argument passed.

```
"""
Generate a mitmproxy dump file.

This script demonstrates how to generate a mitmproxy dump file,
as it would also be generated by passing `-w` to mitmproxy.
In contrast to `-w`, this gives you full control over which
flows should be saved and also allows you to rotate files or log
to multiple files in parallel.
"""
import random
import sys
from typing import BinaryIO

from mitmproxy import http
from mitmproxy import io
import logging

class Writer:
    def __init__(self, path: str) -> None:
        logging.info("Flow writing path is : " + path)
        self.f: BinaryIO = open(path, "wb")
        self.w = io.FlowWriter(self.f)

    def response(self, flow: http.HTTPFlow) -> None:
        if random.choice([True, False]):
            self.w.add(flow)

    def done(self):
        self.f.close()


addons = [Writer(sys.argv[1])]

```

Trying to pass argument to the script
```
# mitmdump -s "/foo/bar/flow-writer.py /foo/bar/myflow.mitm"
/usr/local/bin/mitmdump: No such script: /foo/bar/flow-writer.py /foo/bar/myflow.mitm
```

Loading the script without arguments, my logger shows also that `sys.argv[1]` may not be the right way to grab the argument?
```
mitmdump -s "/foo/bar/flow-writer.py"
[22:29:28.711] Loading script /foo/bar/flow-writer.py
[22:29:28.714] Flow writing path is : -s
```

**Comment by mhils** [comment]
Thanks for filing this! The example addon here is outdated, this should really be using an option. See https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/options-simple.py and https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/options-configure.py for how that works.


I opened https://github.com/mitmproxy/mitmproxy/issues/6445. :)

**Comment by kartheekgade** [comment]
Hi, I'm able generate dump file in the specified location using arguments.
cmd: mitmproxy -s io-write-flow-file.py -w "path\to\generate\mitmproxy_dump.txt"
I hope this may help.
